app/scripts/modules/core/src/serverGroup/details/ServerGroupDetails.tsx

```diff
@@ -1,85 +1,122 @@
+import { CloudProviderLogo } from 'core/cloudProvider/CloudProviderLogo';
+import { EntityNotifications } from 'core/entityTag/notifications/EntityNotifications';
+import { ReactInjector } from 'core/reactShims';
+import { SETTINGS } from 'core/config/settings';
+import { Spinner } from 'core/widgets/spinners/Spinner';
+import { timestamp } from 'core/utils/timeFormatters';
 
-
-export interface IServerGroupDetailsStateParams {
-  provider: string;
-  accountId: string;
-  region: string;
-  serverGroup: string;
-}
-
-export interface IServerGroupDetailsProps {
-  app: Application;
-  serverGroup: {
-    name: string;
-    accountId: string;
-    region: string;
-  };
-}
-
-export interface IServerGroupDetailsState {
-  angular: {
-    template: string,
-    controller: string,
-  };
-  ServerGroupDetailsComponent: React.ComponentClass<IServerGroupDetailsProps>,
-  accountId: string;
-  provider: string;
-  region: string;
-}
+import { IServerGroupDetailsProps, IServerGroupDetailsState } from './ServerGroupDetailsWrapper';
+import { RunningTasks } from './RunningTasks';
+import { ServerGroupInsightActions } from './ServerGroupInsightActions';
 
 @BindAll()
 export class ServerGroupDetails extends React.Component<IServerGroupDetailsProps, IServerGroupDetailsState> {
+  private destroy$ = new Subject();
+
   constructor(props: IServerGroupDetailsProps) {
     super(props);
 
-    const $stateParams: IServerGroupDetailsStateParams = ReactInjector.$stateParams as any;
-
     this.state = {
-      accountId: $stateParams.accountId,
-      provider: $stateParams.provider,
-      region: $stateParams.region,
-      angular: {
-        template: undefined,
-        controller: undefined,
-      },
-      ServerGroupDetailsComponent: undefined,
+      loading: true,
+      serverGroup: undefined,
+      disabledDate: undefined,
     };
   }
 
+  private autoClose(): void {
+    ReactInjector.$state.params.allowModalToStayOpen = true;
+    ReactInjector.$state.go('^', null, { location: 'replace' });
+  }
+
   public componentDidMount(): void {
-    const { provider, accountId } = this.state;
-    const { versionedCloudProviderService } = ReactInjector;
-    $q.all([
-      versionedCloudProviderService.getValue(provider, accountId, 'serverGroup.detailsComponent'),
-      versionedCloudProviderService.getValue(provider, accountId, 'serverGroup.detailsTemplateUrl'),
-      versionedCloudProviderService.getValue(provider, accountId, 'serverGroup.detailsController'),
-    ]).then((values: [React.ComponentClass<IServerGroupDetailsProps>, string, string]) => {
-      const [ component, templateUrl, controller ] = values;
-      const template = templateUrl ? $templateCache.get<string>(templateUrl) : undefined;
-      this.setState({ angular: { template, controller }, ServerGroupDetailsComponent: component });
-    });
+    this.props.detailsGetter(this.props, (state) => this.setState(state), this.autoClose);
```

Comment:
is `detailsGetter` expected to invoke the state callback sync or async?  If async, this should go in a cancellable Rx chain to avoid calling `setState` on an unmounted component.

Reply:
yes